### PR TITLE
core: add tests for non-zero blockchain start

### LINKED ISSUE #1577: Add tests for non-zero blockchain start
There are a lot of things that can happen during node startup when starting from an already existing DB state (see `(*Blockchain).init`), we need to test for them (and extend `InitMPT` testing at the same time).

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #1577.

### CODE REVIEW

File: pkg/core/blockchain_test.go [hunk 1]
```diff
@@ -2003,3 +2004,157 @@ func setSigner(tx *transaction.Transaction, h util.Uint160) {
 		Scopes:  transaction.Global,
 	}}
 }
+
+func TestBlockchain_StartFromExistingDB(t *testing.T) {
+	var stores = map[string]func(testing.TB, string) (storage.Store, string){
+		"BoltPS":  newBoltStoreForTestingWithPath,
+		"LevelPS": newLevelDBForTestingWithPath,
```
Comment (roman-khimov) [inline pkg/core/blockchain_test.go:2011]: Do we need to test two DBs here? The logic we test is a layer above all of this, any DB backend should work the same.

Comment (AnnaShaleva) [reply pkg/core/blockchain_test.go]: Right, fixed.

File: pkg/core/blockchain_test.go [hunk 2]
```diff
@@ -2003,3 +2005,258 @@ func setSigner(tx *transaction.Transaction, h util.Uint160) {
...
+	var (
+		managementID             = -1
+		managementContractPrefix = 8
+	)
+
+	bc.Close() // Ensure persist is done and persistent store is properly closed.
+
+	t.Run("mismatch storage version", func(t *testing.T) {
+		ps, _ = newPS(t, path)
+		t.Cleanup(func() { require.NoError(t, ps.Close()) })
```
Comment (fyrchik) [inline pkg/core/blockchain_test.go:2037]: We always perform this cleanup in this case and in other tests, move it inside the function?

**Review by roman-khimov — COMMENTED** [review]
Looks good, but can be slightly improved based on @fyrchik's feedback.

File: pkg/core/blockchain_test.go [hunk 3]
```diff
@@ -2003,3 +2005,258 @@ func setSigner(tx *transaction.Transaction, h util.Uint160) {
...
+		})
+		require.Error(t, err)
+		require.True(t, strings.Contains(err.Error(), "P2PSigExtensions setting mismatch"))
+	})
+	t.Run("mismatch P2PStateExchangeExtensions", func(t *testing.T) {
+		ps, _ = newPS(t, path)
+		t.Cleanup(func() { require.NoError(t, ps.Close()) })
+		_, err := initTestChainNoCheck(t, ps, func(c *config.Config) {
+			customConfig(c)
+			c.ProtocolConfiguration.StateRootInHeader = true
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply pkg/core/blockchain_test.go:2072]: It is done intentionally in case if one day we'll extend custom configuration with several settings. Then only one setting is being overriden in this test, while others should stay the same.

**Review by roman-khimov — APPROVED** [review]

**Review by fyrchik — APPROVED** [review]
